Crash report 3.8: NPE in StringUtils.getMd5IntHash

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.reader.ReaderPostListActivity}: java.lang.NullPointerException
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2121)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2146)
       at android.app.ActivityThread.access$700(ActivityThread.java:140)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1238)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:4944)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:511)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1038)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:805)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.NullPointerException
       at org.wordpress.android.util.StringUtils.getMd5IntHash(StringUtils.java:80)
       at org.wordpress.android.util.StringUtils.getMd5Hash(StringUtils.java:90)
       at org.wordpress.android.util.GravatarUtils.blavatarFromUrl(GravatarUtils.java:79)
       at org.wordpress.android.util.GravatarUtils.blavatarFromUrl(GravatarUtils.java:76)
       at org.wordpress.android.ui.reader.views.ReaderBlogInfoView.showBlogInfo(ReaderBlogInfoView.java:118)
       at org.wordpress.android.ui.reader.views.ReaderBlogInfoView.loadFeedInfo(ReaderBlogInfoView.java:56)
       at org.wordpress.android.ui.reader.ReaderPostListFragment.loadBlogOrFeedInfo(ReaderPostListFragment.java:1274)
       at org.wordpress.android.ui.reader.ReaderPostListFragment.onActivityCreated(ReaderPostListFragment.java:444)
       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:848)
       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1035)
       at android.app.BackStackRecord.run(BackStackRecord.java:635)
       at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1399)
       at android.app.Activity.performStart(Activity.java:5197)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2094)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2146)
       at android.app.ActivityThread.access$700(ActivityThread.java:140)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1238)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:4944)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:511)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1038)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:805)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/AddQuickPressShortcutActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/AddQuickPressShortcutActivity.java
@@ -86,7 +86,11 @@ public class AddQuickPressShortcutActivity extends ListActivity {
                 accountUsers[validBlogCtr] = curHash.get("username").toString();
                 accountIDs[validBlogCtr] = (Integer)curHash.get("id");
                 String url = curHash.get("url").toString();
-                blavatars[validBlogCtr] = GravatarUtils.blavatarFromUrl(url, 60);
+                if (url != null) {
+                    blavatars[validBlogCtr] = GravatarUtils.blavatarFromUrl(url, 60);
+                } else {
+                    blavatars[validBlogCtr] = "";
+                }
                 accountNames.add(validBlogCtr, blogNames[i]);
                 validBlogCtr++;
             }

--- a/WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderBlogInfoView.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderBlogInfoView.java
@@ -111,11 +111,13 @@ public class ReaderBlogInfoView extends FrameLayout {
 
         int imageResId = (blogInfo.isExternal() ? R.drawable.gravatar_placeholder : R.drawable.blavatar_placeholder);
         int imageSz = getResources().getDimensionPixelSize(R.dimen.avatar_sz_medium);
-        String imageUrl;
+        String imageUrl = "";
         if (blogInfo.hasImageUrl()) {
             imageUrl = GravatarUtils.fixGravatarUrl(blogInfo.getImageUrl(), imageSz);
         } else {
-            imageUrl = GravatarUtils.blavatarFromUrl(blogInfo.getUrl(), imageSz);
+            if (blogInfo.getUrl() != null) {
+                imageUrl = GravatarUtils.blavatarFromUrl(blogInfo.getUrl(), imageSz);
+            }
         }
         imgBlavatar.setErrorImageResId(imageResId);
         imgBlavatar.setImageUrl(imageUrl, ImageType.BLAVATAR);

